### ISSUE TITLE
Suppress session overwrite on certain routes

### ISSUE THREAD

**Issue by DaveCTurner** [yesodweb/yesod#1292]
We have some long-polled routes that sit in a Yesod handler for an extended period (~30s) before returning. We have an issue where if the session is changed during the wait (e.g. due to actions in another tab) then the changes are overwritten when the long poll returns and resets the session cookie according to its stale view of the world.

Is there a way to suppress the setting of the `_SESSION` cookie from within certain handlers? Or is there a better way to achieve the result we are after?

**Comment by snoyberg** [comment]
I'm not aware of such functionality existing, but I don't see a problem with adding it. It should be a matter of just adding another method to the `Yesod` typeclass. Interested in trying out a PR that solves your problem?

**Comment by DaveCTurner** [comment]
Should be able to submit a PR although can't promise it'll be any time soon.

In the meantime we have managed a workaround: since we already have a (slightly) custom `SessionBackend`, we've just modified the `SaveSession` thing it returns to recognise a specific key in the map and return an empty list if it's present. Not the most elegant solution, sure, but it'll do for now.

https://hackage.haskell.org/package/yesod-core-1.4.22/docs/Yesod-Core-Types.html
